.github/workflows/helm-publish.yml

```diff
@@ -34,6 +36,21 @@ jobs:
           echo "X_TAG=${X_TAG}" >> $GITHUB_ENV
           echo "tag=${X_TAG}" >> $GITHUB_OUTPUT
 
+      - name: Check release type
+        id: check_release
+        run: |
+          TYPE="production"
+          if [[ ${TAG} =~ "rc" ]]; then
```

Comment:
1. Looks like `X_TAG` is used in this workflow, not `TAG` 
2. How `X_TAG` will look when we use rc? X_TAG comes as Helm version to `Publish Helm charts` step

Reply:
Thanks, that was a typo

I've rewritten the changes as you requested, please look once more